### PR TITLE
feat(smart): Entrypoint v0.7 support

### PR DESCRIPTION
**Pull request by joaquim-verges**
## Problem solved

passing entrypoint 0.7 + 0.7 compatible factory is now supported

---

## PR-Codex overview
This PR adds support for Entrypoint v0.7 for smart accounts. 

### Detailed summary
- Updated smart account handling for v0.7
- Improved factory usage in smart wallet creation
- Refactored default account factory constants
- Enhanced event handling for `PostOpRevertReason`
- Improved paymaster handling in smart wallets

> The following files were skipped due to too many changes: `packages/thirdweb/src/wallets/smart/lib/packUserOp.ts`, `packages/thirdweb/src/wallets/smart/index.ts`, `packages/thirdweb/src/wallets/smart/types.ts`, `packages/thirdweb/src/wallets/smart/lib/bundler.ts`, `packages/thirdweb/src/extensions/erc4337/__generated__/IEntryPoint_v07/read/getUserOpHash.ts`, `packages/thirdweb/src/wallets/smart/smart-wallet-integration-v07.test.ts`, `packages/thirdweb/src/wallets/smart/lib/userop.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by joaquim-verges** [comment]
* **#4106** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4106?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joaquim-verges and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/wallets/smart/lib/userop.ts [hunk 1]
```diff
@@ -113,24 +118,88 @@
...
+  const [isDeployed, callData, gasFees, nonce] = await Promise.all([
+    isContractDeployed(accountContract),
+    encode(executeTx),
+    getGasFees({
+      executeTx,
+      bundlerOptions,
+      chain,
+      client,
+    }),
+    getAccountNonce({
```
Comment (gregfromstl) [inline packages/thirdweb/src/wallets/smart/lib/userop.ts:141]: What does this do if the account isn't deployed?

Comment (gregfromstl) [inline packages/thirdweb/src/wallets/smart/lib/userop.ts:141]: What does this do if the account isn't deployed?

File: packages/thirdweb/src/wallets/smart/lib/userop.ts [hunk 2]
```diff
@@ -191,7 +419,10 @@ export async function createUnsignedUserOp(args: {
       entrypointAddress: overrides?.entrypointAddress,
```
Comment (gregfromstl) [inline packages/thirdweb/src/wallets/smart/lib/userop.ts:419]: <i class='graphite__hidden'>[Re: line +415]<br/><br/></i>

Could we call this `paymasterEstimateResult` or something like that? the result and result2 names are really hard to parse at first<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4106?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: packages/thirdweb/src/wallets/smart/lib/utils.ts [hunk 3]
```diff
@@ -32,14 +37,16 @@ export const randomNonce = () => {
...
-export function hexlifyUserOp(userOp: UserOperation): UserOperationHexed {
+export function hexlifyUserOp(
+  userOp: UserOperationV06 | UserOperationV07,
+): UserOperationV06Hexed | UserOperationV07Hexed {
   return Object.fromEntries(
     Object.entries(userOp).map(([key, val]) => [
       key,
       // turn any value that's not hex into hex
-      isHex(val) ? val : toHex(val),
+      val === undefined ? undefined : isHex(val) ? val : toHex(val),
```
Comment (gregfromstl) [inline packages/thirdweb/src/wallets/smart/lib/utils.ts:47]:
```suggestion
      (val === undefined || isHex(val)) ? val : toHex(val),
```

**Review by gregfromstl — COMMENTED** [review]
TS stuff looks good, Firekeeper should probably review too for the AA POV

**Review by 0xFirekeeper — APPROVED** [review]

File: packages/thirdweb/src/wallets/smart/lib/userop.ts (hunk 1 above)
Comment (joaquim-verges) [reply packages/thirdweb/src/wallets/smart/lib/userop.ts:141]: this actually fetches the nonce from the Entrypoint, so its always available

**Pull request merged**